### PR TITLE
fixes giving silicons the incorrect survive objective when done manually

### PR DESCRIPTION
**Pull request by Intigracy**

### CODE REVIEW

File: code/datums/mind.dm [hunk 1]
```diff
@@ -495,8 +495,12 @@
 				new_objective.owner = src
 
 			if ("survive")
-				new_objective = new /datum/objective/survive
-				new_objective.owner = src
+				if(issilicon(src))
```
Comment (PJB3005) [inline code/datums/mind.dm]: Alternative:

``` DM
new_objective = new (issilicon(src) ? /datum/objective/siliconsurvive : /datum/objective/survive)
new_objective.owner = src
```

**Comment by ComicIronic** [comment]
Typechecks on src?
VERBODEN

**Comment by Intigracy** [comment]
For what reason?

**Comment by ComicIronic** [comment]
Because it's not OO. If you're typechecking src, you're actually coding behaviour that should be going in a proc override.

**Comment by Intigracy** [comment]
Had to fix it anyway because I was supposed to be checking the current
rather than src.

**Comment by clusterfack** [comment]
Have a mob proc return the correct type intigracy

**Comment by ComicIronic** [comment]
RIP this PR, will do it myself.

**Pull request closed without merging**